### DISCUSSION TITLE
Seq.json persistence with Docker

### DISCUSSION THREAD

**Discussion by fcapolunghi** [datalust/seq-tickets#1223] [Help] [answered]
Hi, 

I've been using the docker container and I am overall pleased with it.

One problem that I have is that Seq.json doesn't seem to persist.
listenUris always reverts to the default list when the container starts.

` "listenUris": [
      "http://localhost:80",
      "http://localhost:5341"
    ]`

Seq data seem to persist nicely.
Using "latest" from about a month ago, on Azure ACI with volume mapped to an Azure file storage.

Am I missing something ?

**Comment by nblumhardt** [comment, marked as answer]
Hi! Yes, the container sets these to the above values on start-up; we took this approach because it will prevent breakage as the container gains additional capabilities/uses additional ports.

I believe it _is_ possible to override these by setting `SEQ_API_LISTENURIS` in the environment, though if you can avoid it, I'd recommend sticking with the defaults.

Unrelated, but in case you missed some details in the docs, we have observed some data corruption issues when Seq is used on Azure Files (which is SMB-based) under load. It should be fine for light use, but if you want to make it more robust you can:

 * Use Azure Premium Files (also SMB-based, I believe, but appears to work more reliably with Seq), and
 * Use an Azure SQL Database for Seq's metadata storage (users, signals, etc. - more in docs.datalust.co/docs/sql-server ).

High-load deployments in Azure are still best on AKS with Azure Disks, or a VM with attached disks.

Hope this all helps!
Nick

**Reply by fcapolunghi** [reply]
This makes sense, I should be able to avoid hacking the environment indeed.
Thank you very much for your answer, and in general for the great work you put into this product, I'm a big fan!

**Reply by nblumhardt** [reply]
Awesome, thank you, @fcapolunghi 👍
